src/App.js

```diff
@@ -1,10 +1,78 @@
-import Products from './components/ProductCardList/Products'
+import { useState } from 'react'
+import { Route, Switch } from 'react-router'
+import { BrowserRouter as Router } from 'react-router-dom'
+import ProductForm from './components/ProductForm/ProductForm'
+import Products from './components/Products/Products'
+
+function App({ initialData }) {
+  const [productData, setProductData] = useState(() => {
+    if (localStorage.getItem('productsLocalStorage')) {
```

Comment:
I don't know if it's that critical but consider that a localStorage variable can exist without a value (e.g the user deleted its content, but not the variable itself). In that case, you'd get an empty string.

I'd pass another condition to the localStorage like this:
```suggestion
    if (localStorage.getItem('productsLocalStorage') !== "") {
```

Reply:
Unfortunately this did not work for me. Also, I learned that I don't need this as an empty string is already falsy.